### PR TITLE
VORTEX-5707: Refactoring analysis support classes.

### PR DESCRIPTION
**Pull request by rkausch**
Fixes #VORTEX-5707

## Proposed Changes
  - Added ArithmeticUtilities to perform basic arithmetic on generic java.lang.Number classes
  - Removed several functions from analysis package
  - Added column function factory, and placed a reference into AnalysisToolbox for other plugins to use

### CODE REVIEW

File: open-sphere-base/analysis/src/main/java/io/opensphere/analysis/table/functions/ColumnFunctionFactory.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import java.util.List;
+
+import io.opensphere.core.Toolbox;
+import io.opensphere.core.util.collections.New;
+
+/**
+ * A factory use to instantiate column functions.
+ */
+public class ColumnFunctionFactory
```
Comment (roddyaj) [inline open-sphere-base/analysis/src/main/java/io/opensphere/analysis/table/functions/ColumnFunctionFactory.java]: I think this is a registry not a factory.

File: open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/ArithmeticUtilities.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    /**
+     * Calculates the sum of the supplied numbers, and returns the result.
+     *
+     * @param leftNumber the left number portion of the sum.
+     * @param rightNumber the right number portion of the sum.
+     * @return the calculated sum.
+     */
+    public static Number sum(Number leftNumber, Number rightNumber)
```
Comment (roddyaj) [inline open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/ArithmeticUtilities.java]: Could this take Number... as arguments instead?

File: open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/ArithmeticUtilities.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+     * @param left the left number portion of the product.
+     * @param right the right number portion of the product.
+     * @return the calculated product.
+     */
+    public static Number product(Number left, Number right)
+    {
+        Number returnValue = null;
+        if (left instanceof Double || left instanceof Float || right instanceof Double || right instanceof Float)
+        {
+            returnValue = new Double(left.doubleValue() * right.doubleValue() + 0.0);
```
Comment (roddyaj) [inline open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/ArithmeticUtilities.java]: Why + 0.0?

Comment (smithtb) [reply open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/ArithmeticUtilities.java]: I was wondering this as well

File: open-sphere-base/analysis/src/main/java/io/opensphere/analysis/table/functions/ColumnFunctionFactory.java (hunk 1 above)
Comment (rkausch) [reply open-sphere-base/analysis/src/main/java/io/opensphere/analysis/table/functions/ColumnFunctionFactory.java]: pe·dan·tic
========
/pəˈdan(t)ik/
*adjective*
 : narrowly, stodgily, and often ostentatiously learned

*ex: a pedantic insistence that we follow the rules exactly*

Comment (rkausch) [reply open-sphere-base/analysis/src/main/java/io/opensphere/analysis/table/functions/ColumnFunctionFactory.java]: Also, done.

File: open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/ArithmeticUtilities.java (hunk 2 above)
Comment (rkausch) [reply open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/ArithmeticUtilities.java]: Done.

File: open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/ArithmeticUtilities.java (hunk 3 above)
Comment (rkausch) [reply open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/ArithmeticUtilities.java]: There's cases that Java will return a -0, due to it's implementation of IEEE Floating Point. Adding zero in this case causes the negative zero to return as positive zero.

File: open-sphere-base/analysis/src/main/java/io/opensphere/analysis/table/functions/ColumnFunctionFactory.java (hunk 1 above)
Comment (roddyaj) [reply open-sphere-base/analysis/src/main/java/io/opensphere/analysis/table/functions/ColumnFunctionFactory.java]: Factory method pattern : Define an interface for creating an object, but let subclasses decide which class to instantiate. The Factory method lets a class defer instantiation it uses to subclasses.

Registry : A well-known object that other objects can use to find common objects and services.

**Pull request merged**